include/z64bgcheck.h

```diff
@@ -87,11 +87,94 @@ typedef struct {
     // 0x0000_00FF = BgCam Index
 } WaterBox; // size = 0x10
 
+typedef enum {
+    /*  0 */ SURFACETYPE_FLOOR_TYPE_0,
+    /*  1 */ SURFACETYPE_FLOOR_TYPE_1,
+    /*  2 */ SURFACETYPE_FLOOR_TYPE_2,
+    /*  3 */ SURFACETYPE_FLOOR_TYPE_3,
+    /*  4 */ SURFACETYPE_FLOOR_TYPE_4,
+    /*  5 */ SURFACETYPE_FLOOR_TYPE_5,
+    /*  6 */ SURFACETYPE_FLOOR_TYPE_6,
+    /*  7 */ SURFACETYPE_FLOOR_TYPE_7,
+    /*  8 */ SURFACETYPE_FLOOR_TYPE_8,
+    /*  9 */ SURFACETYPE_FLOOR_TYPE_9,
+    /* 10 */ SURFACETYPE_FLOOR_TYPE_10,
+    /* 11 */ SURFACETYPE_FLOOR_TYPE_11,
+    /* 12 */ SURFACETYPE_FLOOR_TYPE_12
+} SurfaceTypeFloorType;
```

Comment:
Do all of these enums need to be prefixed with `SURFACETYPE_`? I understand that they are packed in the "surface type data", but I feel like the values themselves could make sense as their own enums. For example I think it could be fine to just have a `FLOOR_TYPE_` or `WALL_TYPE_` enum

Reply:
I guess it can go if it doesn't clash with anything. More opinions?

Reply:
yea I think itll be fine, they are pretty clearly denoted by their own prefix and dont need to be grouped with the variable they are packed into in this case

Reply:
Among all those I think the only ones that would need a more specific prefix are the sfx types, which could probably be `SURFACE_SFX_TYPE_`